kotlin-styled-next/src/main/kotlin/styled/StyleSheet.kt

```diff
@@ -73,18 +74,29 @@ open class StyleSheet(
      *
      * @return A prepared [RuleSet] ready to be used.
      * */
-    internal fun <T : HasCssSuffix> prepareCachedRuleSet(
+    internal fun <T : Any> prepareCachedRuleSet(
         staticCssSuffix: String,
         builder: CssBuilder.(T) -> Unit,
         argument: T
     ): RuleSet {
-        val fullCssSuffix = "$staticCssSuffix-${argument.cssSuffix}"
+        val fullCssSuffix = "$staticCssSuffix-${getArgumentCssSuffix(argument)}"
         return dynamicHolders.getOrPut(fullCssSuffix) {
             DynamicCssHolder(this, fullCssSuffix, { builder.invoke(this, argument) })
                 .also { it.markToInject() }
         }.provideRuleSet()
     }
 
+    private fun getArgumentCssSuffix(argument: Any): String = when (argument) {
+        is Boolean -> argument.toString()
+        is Number -> argument.toString().replace(".", "-")
+        is String -> argument.revampCssSuffix()
+        is Enum<*> -> argument.name.revampCssSuffix()
+        is HasCssSuffix -> argument.cssSuffix.revampCssSuffix()
+        else -> throw IllegalArgumentException("type is unsupported")
+    }
+
+    private fun String.revampCssSuffix() = CSS.escape(this.replace(" ", ""))
```

Comment:
`this.replace(" ", "")` is not necessary, escape does all the necessary escaping.

`revamp` -> `escape`